### PR TITLE
Fix absolute URLs

### LINKED ISSUE #787: Change absolute URLs to APIs to use the xref syntax
There are several topics that are using absolute URLs to docs instead of using the API linking syntax, such as this topic:
https://github.com/dotnet/core-docs/blob/master/docs/standard/collections/index.md

The following topic shows an example of how to use the @ syntax:
https://raw.githubusercontent.com/dotnet/core-docs/master/api/index.md

What follows the @ symbol, it's the API UID. An explanation of the UID and how to use it can be found here: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#cross-reference

Right now, there is no easy way to find the UIDs. The best way to confirm what's the right one is by searching for the type in this repo: https://github.com/docascode/coreapi

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #787 

- Fixed API links that were not using xref
- Fixed links that were going to docs.microsoft.com/dotnet/* insteaf of an .md file

Kept the ones that were being displayed on e-books as-is.

@rpetrusha this includes some of the include files for the migration guide. There were a couple of broken links in there that this will fix.

### CODE REVIEW

File: includes/migration-guide/retargeting/jit/new-64-bit-jit-compiler-net-framework-46.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
...uration file. The following disables compilation with the new 64-bit JIT compiler and instead uses the legacy 64-bit JIT compiler.</li></ul><pre><code class="language-xml">&lt;?xml version =&quot;1.0&quot;?&gt;&#13;&#10;&lt;configuration&gt;&#13;&#10;&lt;runtime&gt;&#13;&#10;&lt;useLegacyJit enabled=&quot;1&quot; /&gt;&#13;&#10;&lt;/runtime&gt;&#13;&#10;&lt;/configuration&gt;&#13;&#10;</code></pre><ul><li>On a per-user basis, you can add a <code>REG_DWORD</code> value named <code>useLegacyJit</code> to the <code>HKEY_CURRENT_USER\SOFTWARE\Microsoft\.NETFramework</code> key of the registry. A value of 1 enables the legacy 64-bit JIT compiler; a value of 0 disables it and enables the new 64-bit JIT compiler.</li><li>On a per-machine basis, you can add a <code>REG_DWORD</code> value named <code>useLegacyJit</code> to the <code>HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework</code> key of the registry. A value of <code>1</code> enables the legacy 64-bit JIT compiler; a value of <code>0</code> disables it and enables the new 64-bit JIT compiler.</li></ul>You can also let us know about the problem by reporting a bug on [Microsoft Connect](https://connect.microsoft.com/VisualStudio).|
```
Comment (rpetrusha) [inline includes/migration-guide/retargeting/jit/new-64-bit-jit-compiler-net-framework-46.md]: Again unrelated to your PR, on the ninth line from the top, could you change the [\<] to [\\\<useLegacyJit>]? The link has dropped out of the build.

File: includes/migration-guide/retargeting/networking/default-value-servicepointmanagersecurityprotocol.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
...
+|Suggestion|This change affects applications that target the .NET Framework 4.7 or later versions.If you prefer to use a defined protocol rather than relying on the system default, you can explicitly set the value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property.If this change is undesirable, you can opt out of it by adding a configuration setting to the [<runtime>](~/docs/framework/configure-apps/file-schema/runtime/runtime-element.md) section of your application configuration file. The following example shows both the <code>&lt;runtime&gt;</code> section and the <code>Switch.System.Net.DontEnableSystemDefaultTlsVersions</code> opt-out switch:<pre><code class="language-xml">&lt;runtime&gt;&#13;&#10;&lt;AppContextSwitchOverrides value=&quot;Switch.System.Net.DontEnableSystemDefaultTlsVersions=true&quot; /&gt;&#13;&#10;&lt;/runtime&gt;&#13;&#10;</code></pre>|
```
Comment (rpetrusha) [inline includes/migration-guide/retargeting/networking/default-value-servicepointmanagersecurityprotocol.md]: can you make the same change here? [\<runtime>] has again dropped out of the published topic.

File: includes/migration-guide/retargeting/winforms/accessibility-improvements-windows-forms-controls.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
...
...r did not recieve notification and as a result users would not be informed that the <code>P:System.Windows.Forms.CheckState</code> had been updated.</li><li>The <code>T:System.Windows.Forms.LinkLabel</code> control has changed the way it notifies Narrator of the text of in the control. Previously, Narrator announced this text twice and read &quot;&amp;&quot; symbols as real text even though they are not visible to a user. The duplicated text was removed from the Narrator announcements, as well as unnecessary &quot;&amp;&quot; symbols.</li><li>The <code>T:System.Windows.Forms.DataGridViewCell</code> control types now correctly report the read-only status to Narrator and other assistive technologies.</li><li>Narrator is now able to read the System Menu of child windows in [Multiple-Document Interface]~/docs/framework/winforms/advanced/multiple-document-interface-mdi-applications.md) applications.</li><li>Narrator is now able to read <code>T:System.Windows.Forms.ToolStripMenuItem</code> controls with a <code>P:System.Windows.Forms.ToolStripItem.Enabled</code> property set to <em>false</em>. Previously, Narrator was unable to focus on disabled menu items to read hte content.</li></ul>|
```
Comment (rpetrusha) [inline includes/migration-guide/retargeting/winforms/accessibility-improvements-windows-forms-controls.md]: Broken link: one of the menubar-control-type links doesn't have an .md file extension

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This looks good, @mairaw, and I've incorporated your changes into the microsoft/dotnet repo. The build report shows 8 broken links, but only one broken link causes it; I've indicated it. There are also three changes unrelated to your PR that I'd appreciate it if you could make.

**Comment by mairaw** [comment]
Thanks @rpetrusha. I've addressed your feedback and a couple of other broken links I found. Better to use rich diff to see the changes I think.

**Review by rpetrusha — APPROVED** [review]
LGTM, @mairaw. You can merge when you're ready.

**Pull request merged**